test/test_baseline_resnet.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+
+try:
+    import torch
+    import torch.distributed
+    from gym import spaces
+
+    from habitat_baselines.rl.ddppo.policy.resnet import resnet18, resnet50
+    from habitat_baselines.rl.ddppo.policy.resnet_policy import ResNetEncoder
+
+    baseline_installed = True
+except ImportError:
+    baseline_installed = False
+
+
+def _npobs_dict_to_tensorobs_dict(npobs_dict):
+    result = {}
+    for k, v in npobs_dict.items():
+        result[k] = torch.as_tensor(v).unsqueeze(0)
+    return result
+
+
+@pytest.mark.skipif(
+    not baseline_installed, reason="baseline sub-module not installed"
+)
+@pytest.mark.parametrize(
+    "observation_space",
+    [
+        spaces.Dict(
+            {
+                "rgb_1": spaces.Box(low=0, high=1, shape=(62, 30, 3)),
+                "rgb_2": spaces.Box(low=0, high=1, shape=(62, 30, 2)),
+            },
+        ),
+        spaces.Dict(
+            {
+                "rgb_1": spaces.Box(low=0, high=1, shape=(63, 84, 1)),
+                "depth_1": spaces.Box(low=0, high=1, shape=(63, 84, 2)),
+            },
+        ),
+        spaces.Dict(
+            {
+                "rgb_1": spaces.Box(low=0, high=1, shape=(64, 128, 3)),
+            },
+        ),
+        spaces.Dict(
+            {
+                "rgb_1": spaces.Box(low=0, high=1, shape=(65, 30, 3)),
+                "rgb_2": spaces.Box(low=0, high=1, shape=(65, 30, 1)),
+                "depth_1": spaces.Box(low=0, high=1, shape=(65, 30, 2)),
+            },
+        ),
+        spaces.Dict(
+            {
+                "rgb_1": spaces.Box(low=0, high=1, shape=(66, 64, 3)),
+                "depth_2": spaces.Box(low=0, high=1, shape=(66, 64, 2)),
+            },
+        ),
+    ],
+)
+@pytest.mark.parametrize("backbone", [resnet18, resnet50])
+def test_resnetencoder_initialization(observation_space, backbone):
+    encoder = ResNetEncoder(observation_space, make_backbone=backbone)
+    obs = observation_space.sample()
+    t_obs = _npobs_dict_to_tensorobs_dict(obs)
```

Comment:
```suggestion
    t_obs = TensorDict.from_tree(observation_space.sample())
```

`from_tree` will call `torch.as_tensor` on all leaves. A TensorDict inherits from Dict[str, torch.Tensor] for it'll be interchangeable here.

Reply:
Unfortunately, I need to use my custom _npobs_dict_to_tensorobs_dict because since I am not using a batch, I need to add an extra dimension to get a batch of 1.

Reply:
```
t_obs = TensorDict.from_tree(observation_space.sample()).map(lambda t: t.unsqueeze(0))
```
Ah, I missed that. `TensorDict` does have a `.map` function that would let you do it tho :)